From 11aee2a00189531a47a31f99e98ed704b22afc59 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Sat, 8 Feb 2014 05:09:41 +0000
Subject: [PATCH 111/351] extcon: arizona: Only stop mic if we are declaring
 3-pole

Button detection requires the microphone detection to be running, so
make sure we don't disable it if we are going to declare mic.

Change-Id: I91f072cd58a33462c89cc817796560040ad11bbd
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/extcon/extcon-arizona.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/extcon/extcon-arizona.c b/drivers/extcon/extcon-arizona.c
index 4031be8..935abfd 100644
--- a/drivers/extcon/extcon-arizona.c
+++ b/drivers/extcon/extcon-arizona.c
@@ -879,7 +879,8 @@ static void arizona_micd_detect(struct work_struct *work)
 	if (!(val & ARIZONA_MICD_STS)) {
 		dev_warn(arizona->dev, "Detected open circuit\n");
 		info->mic = arizona->pdata.micd_open_circuit_declare;
-		arizona_stop_mic(info);
+		if (!info->mic)
+			arizona_stop_mic(info);
 		arizona_identify_headphone(info);
 		info->detecting = false;
 		goto handled;
-- 
1.7.9.5

